test/test_genome.py

```diff
@@ -548,3 +548,129 @@ def test_mutate_does_not_reinitialize_parameters(genome_params, rng, mutation_ra
     assert genome._parameter_names_to_values["<p2>"] == pytest.approx(
         parameter_names_to_values_before["<p2>"]
     )
+
+
+def test_genome_reordering_empirically(rng):
+    # empirically test that reordering doesn't change the output function of a genome
+
+    # target: f(x_0) = x_0 ** 2 - x_0 + 1
+
+    genome_params = {
+        "n_inputs": 1,
+        "n_outputs": 1,
+        "n_columns": 10,
+        "n_rows": 1,
+        "levels_back": None,
+        "primitives": (cgp.Mul, cgp.Sub, cgp.Add, cgp.ConstantFloat),
+    }
+
+    genome = cgp.Genome(**genome_params)
+
+    dna_fixed = [
+        ID_INPUT_NODE,
+        ID_NON_CODING_GENE,
+        ID_NON_CODING_GENE,
+        0,  # Mul -> outs x² (address 1)
+        0,  # x
+        0,  # x
+        1,  # Sub -> x² - x (address 2)
+        1,  # x²
+        0,  # x
+        1,  # Sub -> outs 0 (address 3)
+        0,  # x
+        0,  # x
+        3,  # const -> outs 1 (address 4)
+        1,  # address of x² (unused)
+        2,  # address of 0 (unused)
+        3,  # const -> outs 1 (address 5)
+        0,
+        0,
+        2,  # Add -> x² - x + 1 (address 6)
+        2,  # x² - x
+        4,  # 1
+        3,  # const (address 7)
+        0,
+        0,
+        3,  # const (address 8)
+        0,
+        0,
+        3,  # const (address 9)
+        0,
+        0,
+        3,  # const (address 10)
+        0,
+        0,
+        ID_OUTPUT_NODE,
+        6,
+        ID_NON_CODING_GENE,
+    ]
+
+    genome.dna = dna_fixed
+    sympy_expression = cgp.CartesianGraph(genome).to_sympy()
+    n_reorderings = 100
+    for _ in range(n_reorderings):
+        genome.reorder(rng)
+        sympy_expression_after_reorder = cgp.CartesianGraph(genome).to_sympy()
+        assert sympy_expression_after_reorder == sympy_expression
+
+    # target: f(x_0,x_1) = x_0 ** 2 - x_1 + 1
```

Comment:
what's different to the check above? maybe move to separate function

Reply:
One genome has only a single input and the other has two. But i agree it might not be necessary to test the same for two similar `genome_params` configurations. Should I remove one of them?

Reply:
yeah in that case i would just keep the more general test case